types/nodemailer/index.d.ts

```diff
@@ -13,7 +14,18 @@ export type SendMailOptions = Mail.Options;
 
 export type Transporter<T = any> = Mail<T>;
 
-export type SentMessageInfo = any;
+export interface SentMessageInfo {
```

Comment:
Where is this type used? I couldn't find any occurrences of it in this file. Where does it come from, in the original library?
Also, is it related to `SESTransport.SentMessageInfo`?

Please clarify why this change is needed so a maintainer can review it. Thanks.

Reply:
I encountered an issue when using the @nestjs-modules/mailer package, which imports SentMessageInfo from nodemailer. The MailerService.sendMail(sendMailOptions: ISendMailOptions): Promise<SentMessageInfo> method currently defines Promise<SentMessageInfo> as any, even though the actual returned data has a specific structure.

This lack of a proper type definition causes inconvenience because the actual type of the returned data isn't known until runtime, making it harder to work with and test the method. The any type doesn't provide any type safety or IntelliSense support, which can lead to potential issues.

To address this, I noticed that the returned data overlaps with SESTransport.SentMessageInfo, so I used that as a reference to define a more accurate type for SentMessageInfo. This change aims to provide better type safety and improve the developer experience by making the expected structure of the returned data clear during development.

Reply:
I really don't think this change is needed. `SentMessageInfo` is unreferenced, so at best I can only think of it as some sort of common root type for all of the different infos, but that's not how it's used in `createTransport`; `T` can just be `any`.

The props you've added here are pretty much just copies of `SMTPConnection.SentMessageInfo`, but this library works with other transports which will not have the same properties.

I think the only thing I can think would be valuable would be to look at what props the library _actually_ needs out of its transports (theoretically the same as or a subset of all of the different referenced transports), then "bound" `T` everywhere to that type.